django/db/models/fields/related.py

```diff
@@ -1258,6 +1258,16 @@ def _check_ignored_options(self, **kwargs):
                 )
             )
 
+        if self.remote_field.symmetrical and self._related_name:
+            warnings.append(
+                checks.Warning(
+                    f'related_name has no effect on ManyToManyField '
+                    'with a symmetrical relationship.',
```

Comment:
Good points. I put `(e.g. to "self")` because I'm under the impression that you can have a symmetrical relationship to a non-self model as well (probably pretty rare).